frontend/pages/views/PhotographerListView.js

```diff
@@ -3,66 +3,273 @@ import Footer from "../../components/Footer";
+                                }`}
+                                onClick={() => {
+                                    if (this.state.selected !== item) {
+                                        this.setState({selected: item});
+                                    } else {
+                                        this.setState({selected: null});
+                                    }
+                                }}
+                            >
+                                {item}
+                            </div>
+                        );
+                    })}
+                </div>
+            </div>
+        );
+    }
+}
+
 export class PhotographerListView extends React.Component {
     constructor(props) {
         super(props);
+
         this.state = {
-            photographers: JSON.parse(this.props.photographers)
+            photographers: JSON.parse(this.props.photographers),
+            timer: null,
+            activeDropdown: null,
         };
+
+        this.LOCATIONS = ["1", "2", "3"];
+        this.SQUARES = ["4", "5", "6"];
+        this.ALPHABET = ["7", "8", "9"];
+        this.SORTS = ["10", "11", "12"];
     }
 
     hrefFunc(number) {
         return `/photographer/${number}`;
-    };
+    }
 
     srcFunc(number) {
         return `${this.props.photoListDir}/${number}_photo.jpg`;
-    };
+    }
 
     getPhotoList() {
         const photoSize = [100, 100];
         return this.state.photographers.map((photog, k) => {
             return (
                 <li className="col-2 col-lg-2 one-photographer list-inline-item" key={k}>
                     <div className="child">
-                        <a key={k}
-                           href={this.hrefFunc(photog.number)}>
+                        <a key={k} href={this.hrefFunc(photog.number)}>
                             <img
                                 alt={photog.number}
                                 width={photoSize[0]}
-                                src={this.srcFunc(photog.number)}/>
+                                src={this.srcFunc(photog.number)}
+                            />
                         </a>
                         <p>{photog.name ? photog.name : "No Name"}</p>
                     </div>
                 </li>
             );
         });
+    }
 
+    handleScroll = () => {
+        // Detecting scroll end adapted from https://stackoverflow.com/a/4620986
+        if (this.scrollOverTimer !== null) {
+            clearTimeout(this.scrollOverTimer);
+        }
+
+        let dropdowns = document.getElementsByClassName("dropdown-items");
+        for (const elt of dropdowns) {
+            elt.classList.add("d-none");
+        }
+
+        this.scrollOverTimer = setTimeout(() => {
+            // If the timer elapses, show to
+            // this.setState({retract:true});
+            document.getElementsByClassName("banner")[0].classList.add("grow");
```

Comment:
Rather than querying every time for this element, you can save it in a variable `const banner = document.getElementsByClassName("banner")[0]` and then `banner.classList.add("grow")`, etc.

or even snazzier: `document.querySelector(".banner")`

btw -- the fact that it's always the 0th element of a list of elements you expect to be 1-long is a code smell that maybe banner should be an id rather than a classname.